version: '3.9'

services:
  # TODO: resolve caddy issues
  cloudbeaver:
    image: dbeaver/cloudbeaver:latest
    container_name: cloudbeaver
    restart: always
    depends_on:
      caddy: { condition: service_started }
    # ports:
    #   - 8080:8978
    networks:
      - internal
    labels:
      # caddy: dev.vasandani.me
      # caddy.handle_path: /database*
      caddy: db.dev.vasandani.me
      caddy.handle_path: /*
      caddy.handle_path.reverse_proxy: "{{ upstreams 8978 }}"
      caddy.tls: internal
      logging: cloudbeaver
      logging_jobname: containerlogs
    volumes:
      - cloudbeaver_data:/opt/cloudbeaver/workspace
      # - ./provided-connections.json:/opt/cloudbeaver/workspace/GlobalConfiguration/.dbeaver/provided-connections.json
    environment:
      - CB_SERVER_NAME='Test Server'
      # - CB_SERVER_URL=http://0.0.0.0:8978
      - CB_SERVER_URL=http://db.dev.vasandani.me
      - CB_ADMIN_NAME=admin
      - CB_ADMIN_PASSWORD=changeme
    logging:
      driver: fluentd
      options:
        fluentd-address: 127.0.0.1:24224
        fluentd-async-connect: "true"
        tag: service.cloudbeaver

volumes:
  cloudbeaver_data:
